### ISSUE TITLE
Yet another String.split() inconcistency

### ISSUE THREAD

**Issue by sjrd** [scala-js/scala-js#2592]
Scala/JVM:
```scala
scala> "a".split("a")
res12: Array[String] = Array()
```
Scala.js:
```scala
Array("")
```
In the Scala.js implem we have:
```scala
    /* With `limit == 0`, remove trailing empty strings (but do not reduce
     * the array to be zero length).
     */
```
It appears the "but do not reduce the array to be zero length" is erroneous. The array can be shrunk to be empty.

However, splitting an empty string always returns an array of exactly one empty empty string.

**Comment by gzm0** [comment]
Hahahahaha :P It's almost time we started fuzz testing this guy.

**Comment by sjrd** [comment]
Basically the fix to #987, in 662e93ad54c699a7e925368262fdb960110a7dbc, was not the right way to fix the problem. Instead of special-casing the resulting array, we need to special-case an empty input string.
